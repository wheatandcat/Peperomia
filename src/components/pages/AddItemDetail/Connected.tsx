import React, { memo, useCallback, useState } from 'react';
import { Alert } from 'react-native';
import dayjs from 'dayjs';
import advancedFormat from 'dayjs/plugin/advancedFormat';
import 'dayjs/locale/ja';
import {
  useCreateItemDetailMutation,
  NewItem,
  NewItemDetail,
} from 'queries/api/index';
import Plain from './Plain';
import { Props as IndexProps } from './';

dayjs.extend(advancedFormat);

type Props = IndexProps & {
  date: string;
  itemId: string;
  priority: number;
  onCallback: () => void;
};

export type ConnectedType = {
  date: string;
  onDismiss: () => void;
  onSave: (item: NewItem) => void;
  onIcons: (kind: string) => void;
} & State;

type State = {
  selectedKind: string;
};

const Connected: React.FC<Props> = memo((props) => {
  const [state, setState] = useState<State>({ selectedKind: '' });
  const [
    createItemDetailMutation,
    createItemDetailMutationData,
  ] = useCreateItemDetailMutation({
    async onCompleted() {
      await props?.onCallback?.();

      props.navigation.goBack();
    },
    onError(err) {
      Alert.alert('保存に失敗しました', err.message);
    },
  });

  const onSave = useCallback(
    (item: NewItem) => {
      const itemDetail: NewItemDetail = {
        ...item,
        priority: props.priority,
        date: props.date,
        itemId: props.itemId,
      };

      const variables = {
        itemDetail,
      };

      createItemDetailMutation({ variables });
    },
    [createItemDetailMutation, props]
  );

  const onDismiss = useCallback(() => {
    props.navigation.goBack();
  }, [props.navigation]);

  const onIcons = useCallback(
    (kind: string) => {
      props.navigation.navigate('Icons', {
        kind,
        onSelectIcon: (selectedKind: string) => {
          setState((s) => ({
            ...s,
            selectedKind,
          }));
        },
      });
    },
    [props]
  );

  return (
    <Plain
      {...state}
      date={props.date}
      mutationData={createItemDetailMutationData}
      onDismiss={onDismiss}
      onSave={onSave}
      onIcons={onIcons}
    />
  );
});

export default Connected;
