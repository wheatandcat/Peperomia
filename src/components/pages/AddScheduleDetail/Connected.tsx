import React, { memo, useState, useCallback, useEffect } from 'react';
import { Alert } from 'react-native';
import uuidv1 from 'uuid/v1';
import { getKind } from 'peperomia-util';
import { ActionSheetOptions } from '@expo/react-native-action-sheet';
import { ItemDetail, SelectItemDetail } from 'domain/itemDetail';
import { createItemDetail } from 'lib/itemDetail';
import { ContextProps as ItemContextProps } from 'containers/Items';
import { ContextProps as AuthContextProps } from 'containers/Auth';
import Page from 'components/templates/CreateScheduleDetail/Page';
import { Props as IndexProps } from './';

type State = ItemDetail & {
  iconSelected: boolean;
};

type Props = ItemDetail &
  Pick<ItemContextProps, 'itemDetails' | 'refreshData'> &
  Pick<AuthContextProps, 'uid'> & {
    kind: string;
    itemId: string;
    priority: number;
    navigation: IndexProps['navigation'];
    showActionSheetWithOptions: (
      options: ActionSheetOptions,
      callback: (i: number) => void
    ) => void;
  };

const AddScheduleDetailConnected: React.FC<Props> = (props) => {
  const [state, setState] = useState<State>({
    title: props.title || '',
    place: props.place || '',
    url: props.url || '',
    memo: props.memo || '',
    moveMinutes: props.moveMinutes || 0,
    kind: props.kind,
    priority: props.priority,
    iconSelected: false,
  });
  const { kind, itemId, priority } = props;

  useEffect(() => {
    if (!kind) {
      return;
    }

    if (state.kind !== kind) {
      setState((s) => ({
        ...s,
        kind,
        iconSelected: true,
      }));
    }
  }, [kind, state.kind]);

  const onDismiss = useCallback(() => {
    props.navigation.goBack();
  }, [props.navigation]);

  const save = useCallback(() => {
    props.navigation.navigate('Schedule', {
      itemId,
      refresh: uuidv1(),
    });

    if (props.refreshData) {
      props.refreshData();
    }
  }, [props, itemId]);

  const onSave = useCallback(
    async (
      title: string,
      selectedKind: string,
      place: string,
      url: string,
      m: string,
      moveMinutes: number
    ) => {
      const itemDetail: SelectItemDetail = {
        itemId,
        title,
        place,
        url,
        memo: m,
        kind: selectedKind,
        moveMinutes,
        priority: Number(priority),
      };

      const insertID = await createItemDetail(props.uid, itemDetail);
      if (!insertID) {
        Alert.alert('保存に失敗しました');
        return;
      }

      save();
    },
    [priority, itemId, save, props.uid]
  );

  const onIcons = useCallback(
    (title: string) => {
      props.navigation.navigate('Icons', {
        kind: getKind(title),
        onSelectIcon: (selectedKind: string) => {
          props.navigation.navigate('AddScheduleDetail', {
            kind: selectedKind,
          });
        },
        onDismiss: () => {
          props.navigation.navigate('AddScheduleDetail', {});
        },
        photo: false,
      });
    },
    [props.navigation]
  );

  return (
    <Page
      title={state.title}
      kind={state.kind}
      place={state.place}
      url={state.url}
      memo={state.memo}
      moveMinutes={state.moveMinutes}
      onDismiss={onDismiss}
      onSave={onSave}
      onIcons={onIcons}
      showActionSheetWithOptions={props.showActionSheetWithOptions}
      iconSelected={state.iconSelected}
    />
  );
};

export default memo(AddScheduleDetailConnected);
