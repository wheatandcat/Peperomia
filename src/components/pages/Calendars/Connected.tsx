import React, { memo, useCallback } from 'react';
import dayjs from 'dayjs';
import advancedFormat from 'dayjs/plugin/advancedFormat';
import 'dayjs/locale/ja';
import { ContextProps as CalendarsContextProps } from 'containers/Calendars';
import { Props as IndexProps } from './';
import Page from './Page';

dayjs.extend(advancedFormat);

type Props = IndexProps &
  Pick<CalendarsContextProps, 'calendars' | 'setDate'> & {
    loading: boolean;
  };

export type ConnectedType = {
  onCreate: (date: string) => void;
  onCalendar: (date: string) => void;
  onSchedule: (id: string | number, title: string) => void;
};

const Connected: React.FC<Props> = memo((props) => {
  const onCreate = useCallback(
    (date: string) => {
      props.navigation.navigate('CreatePlan', {
        date,
      });
    },
    [props.navigation]
  );

  const onSchedule = useCallback(
    (id: string | number, title: string) => {
      props.navigation.navigate('Schedule', { itemId: id, title });
    },
    [props.navigation]
  );

  const onCalendar = useCallback(
    (date: string) => {
      props.navigation.navigate('Calendar', {
        date: dayjs(date).format('YYYY-MM-DDT00:00:00'),
      });
    },
    [props.navigation]
  );

  return (
    <Page
      calendars={props.calendars}
      setDate={props.setDate}
      loading={props.loading}
      onCreate={onCreate}
      onSchedule={onSchedule}
      onCalendar={onCalendar}
    />
  );
});

export default Connected;
