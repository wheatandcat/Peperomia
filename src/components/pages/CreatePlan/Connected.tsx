import React, { useState, memo, useEffect, useCallback } from 'react';
import { Alert } from 'react-native';
import { ActionSheetOptions } from '@expo/react-native-action-sheet';
import { ContextProps as ItemContextProps } from 'containers/Items';
import { ContextProps as AuthContextProps } from 'containers/Auth';
import { getKind } from 'peperomia-util';
import { useDidMount } from 'hooks/index';
import { createItem } from 'lib/item';
import { createCalendar } from 'lib/calendar';
import Page from 'components/templates/CreatePlan/Page';
import { Props as indexProps } from './';

type State = {
  input: {
    title: string;
    date: string;
  };
  kind: string;
};

export type Props = Pick<
  ItemContextProps,
  'items' | 'refreshData' | 'calendars'
> &
  Pick<AuthContextProps, 'uid'> & {
    date: string;
    kind: string;
    navigation: indexProps['navigation'];
    showActionSheetWithOptions: (
      options: ActionSheetOptions,
      callback: (i: number) => void
    ) => void;
  };

const Connected: React.FC<Props> = (props) => {
  const [state, setState] = useState<State>({
    input: { title: '', date: '' },
    kind: '',
  });
  const { date, kind } = props;

  useDidMount(() => {
    setState((s) => ({
      ...s,
      input: {
        ...s.input,
        date,
      },
    }));
  });

  useEffect(() => {
    if (kind && kind !== state.kind) {
      setState((s) => ({
        ...s,
        kind,
      }));
    }
  }, [kind, state.kind]);

  const onIcons = useCallback(() => {
    props.navigation.navigate('Icons', {
      kind: getKind(state.input.title),
      onSelectIcon: (selectedKind: string) => {
        props.navigation.navigate('CreatePlan', {
          kind: selectedKind,
        });
      },
      onDismiss: () => {
        props.navigation.navigate('CreatePlan', {});
      },
      photo: true,
    });
  }, [props.navigation, state.input.title]);

  const pushCreateSchedule = useCallback(
    (itemId: number | string) => {
      if (props.refreshData) {
        props.refreshData();
      }

      props.navigation.navigate('CreateSchedule', {
        itemId,
        title: state.input.title,
      });
    },
    [props, state.input.title]
  );

  const onInput = useCallback((name: string, value: string) => {
    setState((s) => ({
      ...s,
      input: {
        ...s.input,
        [name]: value,
      },
    }));
  }, []);

  const save = useCallback(
    async (insertId: number | string) => {
      if (state.input.date) {
        // 日付のデータがある場合ははcalendarに登録する
        const calendar = {
          itemId: insertId,
          date: state.input.date,
        };

        const insertID = await createCalendar(props.uid, calendar);

        if (!insertID) {
          Alert.alert('保存に失敗しました');
          return;
        }

        pushCreateSchedule(insertId);
      } else {
        pushCreateSchedule(insertId);
      }
    },
    [pushCreateSchedule, state.input.date, props.uid]
  );

  const onSave = useCallback(async () => {
    if (state.input.date) {
      const check = (props.calendars || []).find(
        (calendar) => calendar.date === state.input.date
      );

      if (check) {
        Alert.alert('同じ日にスケジュールが既に登録されています');
        return;
      }
    }

    const item = {
      title: state.input.title,
      kind: state.kind || getKind(state.input.title),
    };

    const insertID = await createItem(props.uid, item);
    if (!insertID) {
      Alert.alert('保存に失敗しました');
      return;
    }

    save(insertID);
  }, [
    props.calendars,
    save,
    state.input.date,
    state.input.title,
    state.kind,
    props.uid,
  ]);

  const onHome = useCallback(() => {
    props.navigation.goBack();
  }, [props.navigation]);

  return (
    <Page
      mode="new"
      title={state.input.title}
      date={state.input.date}
      kind={state.kind}
      onInput={onInput}
      onSave={onSave}
      onIcons={onIcons}
      onHome={onHome}
      showActionSheetWithOptions={props.showActionSheetWithOptions}
    />
  );
};

export default memo(Connected);
