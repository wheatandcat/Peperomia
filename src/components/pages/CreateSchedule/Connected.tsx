import React, { useState, memo, useEffect, useCallback } from 'react';
import { Alert } from 'react-native';
import { Item } from 'domain/item';
import { SelectItemDetail } from 'domain/itemDetail';
import { useDidMount } from 'hooks/index';
import { getItemByID } from 'lib/item';
import { getItemDetails, updateItemDetail } from 'lib/itemDetail';
import { ContextProps as AuthContextProps } from 'containers/Auth';
import { Props as IndexProps } from './';
import Page from 'components/templates/CreateSchedule/Page';

type Props = Pick<AuthContextProps, 'uid'> & {
  refresh: string;
  itemId: string;
  navigation: IndexProps['navigation'];
};

type State = {
  item: Item;
  itemDetails: SelectItemDetail[];
  refresh: string;
};

const Connected: React.FC<Props> = (props) => {
  const [state, setState] = useState<State>({
    item: { title: '', kind: '' },
    itemDetails: [],
    refresh: '0',
  });

  const setItemDetails = useCallback(
    (data: SelectItemDetail[]) => {
      const priorities = data.map((item) => item.priority);
      const uniquePriorities = priorities.filter(
        (x: number, i: number, self: number[]) => self.indexOf(x) === i
      );

      // priorityが重複していない
      if (priorities.length === uniquePriorities.length) {
        props.navigation.setParams({
          itemDetails: data,
        });
        setState((s) => ({
          ...s,
          itemDetails: data,
        }));

        return;
      }

      // priorityが重複している場合はid順でpriorityをupdateする
      const itemDetails: SelectItemDetail[] = data.map((itemDetail, index) => ({
        ...itemDetail,
        priority: index + 1,
      }));

      itemDetails.forEach(async (itemDetail, index) => {
        const v = {
          ...itemDetail,
          id: itemDetail.id || '',
          priority: index + 1,
        };

        const ok = await updateItemDetail(props.uid, v);
        if (!ok) {
          Alert.alert('保存に失敗しました');
          return;
        }
      });

      setState((s) => ({
        ...s,
        itemDetails: itemDetails,
      }));
    },
    [props.navigation, props.uid]
  );

  const setItem = useCallback((data: Item) => {
    setState((s) => ({
      ...s,
      item: {
        title: data.title,
        kind: data.kind,
      },
    }));
  }, []);

  useDidMount(() => {
    const setItemByItemID = async () => {
      const item = await getItemByID(props.uid, props.itemId);
      setItem(item);
    };

    const setItemDetailsByItemID = async () => {
      const itemDetails = await getItemDetails(props.uid, props.itemId);
      setItemDetails(itemDetails);
    };

    setItemByItemID();
    setItemDetailsByItemID();
  });

  useEffect(() => {
    const setItemDetailsByItemID = async () => {
      const itemDetails = await getItemDetails(props.uid, props.itemId);
      setItemDetails(itemDetails);
    };

    if (props.refresh !== state.refresh) {
      setItemDetailsByItemID().then(() => {
        setState((s) => ({ ...s, refresh: props.refresh }));
      });
    }
  }, [setItemDetails, state.refresh, props.refresh, props.uid, props.itemId]);

  const onCreateScheduleDetail = useCallback(() => {
    props.navigation.navigate('CreateScheduleDetail', {
      itemId: props.itemId,
    });
  }, [props.navigation, props.itemId]);

  const onScheduleDetail = useCallback(
    (id: string) => {
      props.navigation.navigate('ScheduleDetail', {
        itemDetailId: id,
        priority: state.itemDetails.length + 1,
      });
    },
    [props.navigation, state.itemDetails.length]
  );

  const onFinish = useCallback(() => {
    if (state.itemDetails.length === 0) {
      Alert.alert(
        'まだ予定の設定がありません',
        '本当に完了しますか？',
        [
          {
            text: 'キャンセル',
            style: 'cancel',
          },
          {
            text: '完了する',
            onPress: () => {
              props.navigation.navigate('Home', { refresh: true });
            },
          },
        ],
        { cancelable: false }
      );
    } else {
      props.navigation.navigate('Home', { refresh: true });
    }
  }, [props.navigation, state.itemDetails.length]);

  const onGoBack = useCallback(() => {
    if (state.itemDetails.length === 0) {
      Alert.alert(
        'まだ予定の設定がありません',
        '本当に閉じますか？',
        [
          {
            text: 'キャンセル',
            style: 'cancel',
          },
          {
            text: '閉じる',
            onPress: () => {
              props.navigation.navigate('Home', { refresh: true });
            },
          },
        ],
        { cancelable: false }
      );
    } else {
      props.navigation.navigate('Home', { refresh: true });
    }
  }, [props.navigation, state.itemDetails.length]);

  return (
    <Page
      title={state.item.title}
      kind={state.item.kind}
      data={state.itemDetails}
      onScheduleDetail={onScheduleDetail}
      onCreateScheduleDetail={onCreateScheduleDetail}
      onFinish={onFinish}
      onGoBack={onGoBack}
    />
  );
};

export default memo(Connected);
