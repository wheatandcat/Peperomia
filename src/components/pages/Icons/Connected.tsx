import React, { memo } from 'react';
import Page from './Page';

type Props = {
  kind: string;
  onSelectIcon: (kind: string) => void;
};

export type ConnectedType = Props;

const Connected: React.FC<Props> = (props) => {
  return <Page kind={props.kind} onSelectIcon={props.onSelectIcon} />;
};

export default memo(Connected);
