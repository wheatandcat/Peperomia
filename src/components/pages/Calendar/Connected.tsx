import React, { memo, useCallback } from 'react';
import { Alert, Dimensions } from 'react-native';
import { useDeleteCalendarMutation } from 'queries/api/index';
import { ContextProps as CalendarsContextProps } from 'containers/Calendars';
import { copyShareURL } from 'lib/share';
import {
  useUpdateCalendarPublicMutation,
  UpdateCalendarPublicMutationVariables,
} from 'queries/api/index';
import Toast from 'react-native-root-toast';
import useCalendar from 'hooks/useCalendar';
import { Props as IndexProps } from './';
import Plain, { QueryProps } from './Plain';

export type CalendarType = QueryData<QueryProps, 'calendar'>;
export type ItemDetailType = ArrayType<CalendarType['item']['itemDetails']>;

type Props = IndexProps & {
  date: string;
} & Pick<CalendarsContextProps, 'refetchCalendars'>;

export type ConnectedType = {
  onDismiss: () => void;
  onUpdate: () => void;
  onDelete: () => void;
  onShare: (open: boolean) => void;
  onAddItemDetail: () => void;
  onItemDetail: (itemDetailId: string) => void;
  create: boolean;
};

const Connected: React.FC<Props> = memo((props) => {
  const [updateCalendarPublicMutation] = useUpdateCalendarPublicMutation({
    async onCompleted(r) {
      const id = r.updateCalendarPublic.id;
      const open = r.updateCalendarPublic.public;

      if (open) {
        copyShareURL(String(id));
      } else {
        const { height } = Dimensions.get('window');

        let toast = Toast.show('リンクを非公開にしました', {
          duration: Toast.durations.LONG,
          position: height - 150,
          shadow: true,
          animation: true,
          hideOnPress: true,
          delay: 0,
        });

        setTimeout(function () {
          Toast.hide(toast);
        }, 3000);
      }
    },
    onError(err) {
      Alert.alert('公開に失敗しました', err.message);
    },
  });

  const [deleteCalendarMutation] = useDeleteCalendarMutation({
    async onCompleted() {
      await props.refetchCalendars?.();

      props.navigation.popToTop();
    },
    onError(err) {
      Alert.alert('削除に失敗しました', err.message);
    },
  });

  const { data, loading, error, refetch } = useCalendar({
    variables: {
      date: props.date,
    },
    fetchPolicy: 'network-only',
  });

  const onDismiss = useCallback(async () => {
    await props.refetchCalendars?.();

    props.navigation.popToTop();
  }, [props]);

  const onUpdate = useCallback(() => {
    const itemDetail = (data?.calendar?.item?.itemDetails || []).find(
      (v) => v?.priority === 1
    );

    props.navigation.navigate('EditItemDetail', {
      date: props.date,
      itemId: data?.calendar?.item.id || '',
      itemDetailId: itemDetail?.id || '',
      onCallback: async () => {
        await props.refetchCalendars?.();
        await refetch?.();
      },
    });
  }, [props, refetch, data]);

  const onDelete = useCallback(() => {
    const variables = {
      calendar: {
        date: props.date,
      },
    };

    deleteCalendarMutation({ variables });
  }, [deleteCalendarMutation, props.date]);

  const onAddItemDetail = useCallback(() => {
    props.navigation.navigate('AddItemDetail', {
      date: props.date,
      itemId: data?.calendar?.item.id || '',
      priority: (data?.calendar?.item.itemDetails?.length || 0) + 1,
      onCallback: async () => {
        await refetch?.();
      },
    });
  }, [props, data, refetch]);

  const onItemDetail = useCallback(
    (itemDetailId: string) => {
      props.navigation.navigate('ItemDetail', {
        date: props.date,
        itemId: data?.calendar?.item.id || '',
        itemDetailId,
        onCallback: async () => {
          await refetch?.();
        },
      });
    },
    [props, data, refetch]
  );

  const onShare = useCallback(
    async (open: boolean) => {
      const variables: UpdateCalendarPublicMutationVariables = {
        calendar: {
          date: props.date,
          public: open,
        },
      };
      updateCalendarPublicMutation({ variables });
    },
    [props, updateCalendarPublicMutation]
  );

  return (
    <Plain
      data={data}
      loading={loading}
      error={error}
      onDismiss={onDismiss}
      onAddItemDetail={onAddItemDetail}
      onUpdate={onUpdate}
      onDelete={onDelete}
      onItemDetail={onItemDetail}
      onShare={onShare}
      create={props.route.params.create || false}
    />
  );
});

export default Connected;
