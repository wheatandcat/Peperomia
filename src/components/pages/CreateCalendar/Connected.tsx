import React, { memo, useCallback, useState } from 'react';
import { Alert } from 'react-native';
import dayjs from 'dayjs';
import advancedFormat from 'dayjs/plugin/advancedFormat';
import 'dayjs/locale/ja';
import { useCreateCalendarMutation, NewItem } from 'queries/api/index';
import { ContextProps as CalendarsContextProps } from 'containers/Calendars';
import Plain from './Plain';
import { Props as IndexProps } from './';

dayjs.extend(advancedFormat);

type Props = IndexProps & {
  date: string;
} & Pick<CalendarsContextProps, 'refetchCalendars'>;

export type ConnectedType = {
  date: string;
  onDismiss: () => void;
  onSave: (item: NewItem) => void;
  onIcons: (kind: string) => void;
} & State;

type State = {
  selectedKind: string;
};

const Connected: React.FC<Props> = memo((props) => {
  const [state, setState] = useState<State>({ selectedKind: '' });
  const [
    createCalendarMutation,
    createCalendarMutationData,
  ] = useCreateCalendarMutation({
    async onCompleted({ createCalendar }) {
      await props.refetchCalendars?.();

      props.navigation.navigate('Calendar', {
        date: dayjs(createCalendar.date).format('YYYY-MM-DDT00:00:00'),
        create: true,
      });
    },
    onError(err) {
      Alert.alert('保存に失敗しました', err.message);
    },
  });

  const onSave = useCallback(
    (item: NewItem) => {
      const variables = {
        calendar: {
          date: props.date,
          item,
        },
      };

      createCalendarMutation({ variables });
    },
    [createCalendarMutation, props.date]
  );

  const onDismiss = useCallback(() => {
    props.navigation.goBack();
  }, [props.navigation]);

  const onIcons = useCallback(
    (kind: string) => {
      props.navigation.navigate('Icons', {
        kind,
        onCallback: (selectedKind: string) => {
          setState((s) => ({
            ...s,
            selectedKind,
          }));
          props.navigation.goBack();
        },
      });
    },
    [props]
  );

  return (
    <Plain
      {...state}
      date={props.date}
      mutationData={createCalendarMutationData}
      onDismiss={onDismiss}
      onSave={onSave}
      onIcons={onIcons}
    />
  );
});

export default Connected;
