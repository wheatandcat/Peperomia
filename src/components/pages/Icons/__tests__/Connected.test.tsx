import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Connected from '../Connected';

describe('components/pages/Icons/Connected.tsx', () => {
  let wrapper: ShallowWrapper;

  const propsData: any = () => ({
    navigation: {
      setParams: jest.fn(),
      navigate: jest.fn(),
    },
    route: {
      params: {
        onDismiss: jest.fn(),
        onSelectIcon: jest.fn(),
        kind: 'park',
        defaultIcon: false,
        photo: false,
      },
    },
  });

  beforeEach(() => {
    wrapper = shallow(<Connected {...propsData()} />);
  });

  it('正常に表示されている', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
