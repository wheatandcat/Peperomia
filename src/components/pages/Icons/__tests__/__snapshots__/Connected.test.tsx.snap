// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/Icons/Connected.tsx 正常に表示されている 1`] = `
<ConnectedActionSheet
  defaultIcon={false}
  kind="park"
  onSelectIcon={[Function]}
  photo={false}
/>
`;
